llvm/lib/Target/AIE/AIEBaseISelLowering.cpp

```diff
@@ -351,6 +365,23 @@ bool AIEBaseTargetLowering::isEligibleForTailCallOptimization(
   return true;
 }
 
+LLT AIEBaseTargetLowering::getOptimalMemOpLLT(
+    const MemOp &Op, const AttributeList &FuncAttributes) const {
+
+  if (Subtarget.isAIE2()) {
+    if (AlloVecRegMemOps && Op.size() >= 32 && Op.isAligned(Align(4)))
+      return LLT::fixed_vector(16, 16);
+    if (AlloVecRegMemOps && Op.size() >= 16 && Op.isAligned(Align(2)))
+      return LLT::fixed_vector(4, 32);
+    if (Op.size() >= 4 && Op.isAligned(Align(4)))
+      return LLT::scalar(32);
+    if (Op.size() >= 2 && Op.isAligned(Align(2)))
+      return LLT::scalar(16);
+  }
+
+  return LLT();
```

Comment:
I guess this means the legalizer will then have to use single-byte memory operations?

Reply:
Yes, the default behavior is to use byte operations.